### ISSUE TITLE
Fatal error with `doctrine/orm` 3.x

### ISSUE THREAD

**Issue by ddebin** [getsentry/sentry-symfony#806]
### How do you use Sentry?

Sentry SaaS (sentry.io)

### SDK version

3.6.0

### Steps to reproduce

Install `"sentry/sentry-symfony": "^4.0"` along `"doctrine/orm": "^3.0"`

### Expected result

`composer install` works with no error

### Actual result

`composer install` (more specifically composer hook `script cache:clear`) crash with
```
!!  
!!  Fatal error: Declaration of Sentry\SentryBundle\Tracing\Doctrine\DBAL\TracingStatementForV3::bindValue($param, $value, $type = Doctrine\DBAL\ParameterType::STRING): bool must be compatible with Doctrine\DBAL\Driver\Statement::bindValue(string|int $param, mixed $value, Doctrine\DBAL\ParameterType $type): void in /Users/ddebin/Dropbox (Personal)/Dev/VM/com.refurb-tracker.symfony4/vendor/sentry/sentry-symfony/src/Tracing/Doctrine/DBAL/TracingStatementForV3.php on line 20
!!  PHP Fatal error:  Declaration of Sentry\SentryBundle\Tracing\Doctrine\DBAL\TracingStatementForV3::bindValue($param, $value, $type = Doctrine\DBAL\ParameterType::STRING): bool must be compatible with Doctrine\DBAL\Driver\Statement::bindValue(string|int $param, mixed $value, Doctrine\DBAL\ParameterType $type): void in /Users/ddebin/Dropbox (Personal)/Dev/VM/com.refurb-tracker.symfony4/vendor/sentry/sentry-symfony/src/Tracing/Doctrine/DBAL/TracingStatementForV3.php on line 20
```

**Comment by ddebin** [comment]
See [`doctrine/orm` 3.0.0](https://github.com/doctrine/orm/releases/tag/3.0.0)

**Comment by Jean85** [comment]
Are you using DBAL 3 or 4?

**Comment by ddebin** [comment]
> Are you using DBAL 3 or 4?

DBAL 4.x, it's more of a problem with `doctrine/dbal` 4

**Comment by Jean85** [comment]
Return type switched from `bool` to `void`, I don't think we can make the interface crosso compatible... 
```php
bindValue($param, $value, $type = Doctrine\DBAL\ParameterType::STRING): bool
bindValue(string|int $param, mixed $value, Doctrine\DBAL\ParameterType $type): void
```

We need to create another one, and this is probably not even the only BC that we need to handle.

**Comment by ddebin** [comment]
Related to https://github.com/getsentry/sentry-symfony/issues/579

**Comment by cleptric** [comment]
We'll add support for DBAL v4; I can't give you a timeline at this point.

**Comment by tacman** [comment]
When creating a new symfony project with --webapp, composer.json adds "doctrine/orm:^3.0" to the requires section.  I wish instead it were "^2.1||^3.0", and let composer figure out which version to use based on what the dependencies support.  

That would help in this case, as well as for API Platform https://github.com/api-platform/core/issues/6150

**Comment by ste93cry** [comment]
Unfortunately, since Doctrine is an optional dependency of this package, we don't require it in `composer.json` besides for development purposes, hence we have no way to prevent the installation of an unsupported version. This is one of the reasons for which I strongly believe that:

- All tracing features should not be enabled by default (and this is a lost battle...)
- We should move the tracing integrations into separate packages, so that we can declare for real the dependencies they really supports in the composer.json

**Comment by alexislefebvre** [comment]
@ste93cry What about declaring a conflict with the unsupported version, until it is supported?

**Comment by Jean85** [comment]
> @ste93cry What about declaring a conflict with the unsupported version, until it is supported?

That would prevent users to use the whole bundle if they use DBAL 4, which is an issue only if you enable tracing.

**Comment by ddebin** [comment]
Could anyone point out how to disable tracing ? 🙂

**Comment by ste93cry** [comment]
You can disable tracing as a whole using:

```yaml
sentry:
  tracing:
    enabled: false
```

or you can disable tracing specifically for DBAL using:

```yaml
sentry:
  tracing:
    dbal:
      enabled: false
```

**Comment by msphn** [comment]
Until resolved I solved that by going with orm 3.* but constraining dbal.

`
"doctrine/dbal": "3.*",
"doctrine/orm": "^3.0.0",
`

**Comment by ddebin** [comment]
> > @ste93cry What about declaring a conflict with the unsupported version, until it is supported?
> 
> That would prevent users to use the whole bundle if they use DBAL 4, which is an issue only if you enable tracing.

@Jean85 Can't use the whole bundle with DBAL 4, even with tracing disabled.

**Comment by BenMorel** [comment]
> Can't use the whole bundle with DBAL 4, even with tracing disabled.

Indeed, even with the whole bundle disabled, the application still fails with:

> Fatal error: Declaration of Sentry\SentryBundle\Tracing\Doctrine\DBAL\TracingStatementForV3::bindValue($param, $value, $type = Doctrine\DBAL\ParameterType::STRING): bool must be compatible with Doctrine\DBAL\Driver\Statement::bindValue(string|int $param, mixed $value, Doctrine\DBAL\ParameterType $type): void in /app/vendor/sentry/sentry-symfony/src/Tracing/Doctrine/DBAL/TracingStatementForV3.php on line 20

**Comment by ste93cry** [comment]
I believe that at this point, a `conflict` is the only viable solution until support for the newer version is worked on @Jean85

**Comment by cleptric** [comment]
Support for `doctrine/dbal` v4.0 was added in [4.14.0](https://github.com/getsentry/sentry-symfony/releases/tag/4.14.0).
